File: frontend/plugins/location-sync/android/src/main/java/com/pathguard/app/plugin/LocationPoint.java
package com.pathguard.app.plugin;

import com.google.gson.JsonObject;

public class LocationPoint implements Comparable<LocationPoint> {
    public double latitude;
    public double longitude;
    public long timestampMs;
    public String clientId;
    public boolean isRecovered;

    public LocationPoint() {}

    public LocationPoint(double latitude, double longitude, long timestampMs, String clientId) {
        this.latitude = latitude;
        this.longitude = longitude;
        this.timestampMs = timestampMs;
        this.clientId = clientId;
        this.isRecovered = false;
    }

    @Override
    public int compareTo(LocationPoint other) {
        return Long.compare(this.timestampMs, other.timestampMs);
    }

    public JsonObject toJson() {
        JsonObject obj = new JsonObject();
        obj.addProperty("lat", latitude);
        obj.addProperty("lng", longitude);
        obj.addProperty("ts", timestampMs);
        obj.addProperty("cid", clientId);
        obj.addProperty("rec", isRecovered);
        return obj;
    }

    public static LocationPoint fromJson(JsonObject obj) {
        LocationPoint p = new LocationPoint();
        p.latitude = obj.get("lat").getAsDouble();
        p.longitude = obj.get("lng").getAsDouble();
        p.timestampMs = obj.get("ts").getAsLong();
        p.clientId = obj.get("cid").getAsString();
        p.isRecovered = obj.has("rec") && obj.get("rec").getAsBoolean();
        return p;
    }
}

File: frontend/plugins/location-sync/ios/Plugin/LocationPoint.swift
import Foundation

struct LocationPoint: Codable, Comparable {
    let latitude: Double
    let longitude: Double
    let timestampMs: Int64
    let clientId: String
    var isRecovered: Bool
    
    enum CodingKeys: String, CodingKey {
        case latitude = "lat"
        case longitude = "lng"
        case timestampMs = "ts"
        case clientId = "cid"
        case isRecovered = "rec"
    }
    
    init(latitude: Double, longitude: Double, timestampMs: Int64, clientId: String) {
        self.latitude = latitude
        self.longitude = longitude
        self.timestampMs = timestampMs
        self.clientId = clientId
        self.isRecovered = false
    }
    
    static func < (lhs: LocationPoint, rhs: LocationPoint) -> Bool {
        lhs.timestampMs < rhs.timestampMs
    }
    
    static func == (lhs: LocationPoint, rhs: LocationPoint) -> Bool {
        lhs.timestampMs == rhs.timestampMs && lhs.clientId == rhs.clientId
    }
}
